include/aws/s3/private/s3_client_impl.h

```diff
@@ -58,6 +58,32 @@ struct aws_s3_endpoint_options {
 
     /* HTTP port override. If zero, determine port based on TLS context */
     uint16_t port;
+
+    /**
+     * Optional.
+     * Proxy configuration for http connection.
+     */
+    struct aws_http_proxy_options *proxy_options;
+
+    /*
+     * Optional.
+     * Configuration for using proxy from environment variable.
+     * Defaults to true
+     * Only works when proxy_options is not set.
+     */
+    struct proxy_env_var_settings *proxy_ev_settings;
+
+    /**
+     * Optional.
+     * If set to 0, default value is used.
+     */
+    uint32_t connect_timeout_ms;
```

Comment:
nit: from user perspective i hate seeing default value is used. can we mention what that default value is

Reply:
OBJECTION❗️

when we write the exact default value, it inevitably gets tweaked to fix someone's ticket and we forget to fix the N places we copy/pasted that value to docs

I specifically put this in our developer guide ;)

https://github.com/awslabs/aws-crt-python/blob/main/guides/dev/writing-code.md#use-none-as-the-default-value-for-optional-arguments
> In documentation, just say "a default value is used" instead of writing in the actual value, because the odds are good that the documentation will get out of sync with reality.

Reply:
I have removed the exact default value.

Reply:
I agree that it can be a pain to keep comments up to date, but I've also been on the other side of this, where i had to read through a bunch of code to find out what the default value is. Its been especially frustrating with java V1 sdks where you have multiple indirections before you can find the value. Java V2 is improving on this and a lot of classes mention default values. I love that pretty much every CDK options field mentions default value in the comment.

Anyways, this is not a hill im willing to die on right now. If we decide not to put defaults in comments as a team, im happy to disagree and commit.

Reply:
Both options seem good to me. One other possible option might be that we could maintain an appendix of all the default values. It will be easier to update it at one location only.